query_execution/QueryContext.cpp

```diff
@@ -58,7 +59,8 @@ QueryContext::QueryContext(const serialization::QueryContext &proto,
                            const CatalogDatabaseLite &database,
                            StorageManager *storage_manager,
                            const tmb::client_id scheduler_client_id,
-                           tmb::MessageBus *bus) {
+                           tmb::MessageBus *bus)
+    : database_(static_cast<const CatalogDatabase&>(database)) {
```

Comment:
This will break the distributed version where `database` is an instance of `CatalogDatabaseCache`.